File: openc3-cosmos-fakesat/targets/FAKESAT/procedures/ops_suite.py
from openc3.script.suite import Suite, Group

load_utility("FAKESAT/lib/fake_sat.py")


class CollectGroup(Group):
    def script_normal_collect(self):
        print(
            f"Running {Group.current_suite()}:{Group.current_group()}:{Group.current_script()}"
        )
        Group.print("Perform Normal Collect")

        cmd_cnt = tlm("FAKESAT HEALTH_STATUS CMD_ACPT_CNT")
        collect_cnt = tlm("FAKESAT IMAGER COLLECTS")
        cmd("FAKESAT COLLECT with TYPE NORMAL, DURATION 5")
        wait_check(f"FAKESAT HEALTH_STATUS CMD_ACPT_CNT == {cmd_cnt + 1}", 5)
        wait_check(f"FAKESAT IMAGER COLLECTS == {collect_cnt + 1}", 5)
        wait_check("FAKESAT IMAGER COLLECT_TYPE == 'NORMAL'", 5)

    def script_special_collect(self):
        print(
            f"Running {Group.current_suite()}:{Group.current_group()}:{Group.current_script()}"
        )
        Group.print("Perform Special Collect")

        cmd_cnt = tlm("FAKESAT HEALTH_STATUS CMD_ACPT_CNT")
        collect_cnt = tlm("FAKESAT IMAGER COLLECTS")
        cmd("FAKESAT COLLECT with TYPE SPECIAL, DURATION 5")
        wait_check(f"FAKESAT HEALTH_STATUS CMD_ACPT_CNT == {cmd_cnt + 1}", 5)
        wait_check(f"FAKESAT IMAGER COLLECTS == {collect_cnt + 1}", 5)
        wait_check("FAKESAT IMAGER COLLECT_TYPE == 'SPECIAL'", 5)


class ModeGroup(Group):
    def script_safe(self):
        fakesat = FakeSat()
        fakesat.safe()

    def script_checkout(self):
        fakesat = FakeSat()
        fakesat.checkout()

    def script_operate(self):
        fakesat = FakeSat()
        fakesat.operate()


class OpsSuite(Suite):
    def __init__(self):
        self.add_group(CollectGroup)
        self.add_group(ModeGroup)

File: openc3-cosmos-fakesat/targets/FAKESAT/procedures/ops_suite.rb
require 'openc3/script/suite.rb'
load_utility 'FAKESAT/lib/fake_sat.rb'

class CollectGroup < OpenC3::Group
  def script_normal_collect
    puts "Running #{OpenC3::Group.current_suite}:#{OpenC3::Group.current_group}:#{OpenC3::Group.current_script}"
    OpenC3::Group.puts "Perform Normal Collect"

    cmd_cnt = tlm("FAKESAT HEALTH_STATUS CMD_ACPT_CNT")
    collect_cnt = tlm("FAKESAT IMAGER COLLECTS")
    cmd("FAKESAT COLLECT with TYPE NORMAL, DURATION 5")
    wait_check("FAKESAT HEALTH_STATUS CMD_ACPT_CNT == #{cmd_cnt + 1}", 5)
    wait_check("FAKESAT IMAGER COLLECTS == #{collect_cnt + 1}", 5)
    wait_check("FAKESAT IMAGER COLLECT_TYPE == 'NORMAL'", 5)
  end

  def script_special_collect
    puts "Running #{OpenC3::Group.current_suite}:#{OpenC3::Group.current_group}:#{OpenC3::Group.current_script}"
    OpenC3::Group.puts "Perform Special Collect"

    cmd_cnt = tlm("FAKESAT HEALTH_STATUS CMD_ACPT_CNT")
    collect_cnt = tlm("FAKESAT IMAGER COLLECTS")
    cmd("FAKESAT COLLECT with TYPE SPECIAL, DURATION 5")
    wait_check("FAKESAT HEALTH_STATUS CMD_ACPT_CNT == #{cmd_cnt + 1}", 5)
    wait_check("FAKESAT IMAGER COLLECTS == #{collect_cnt + 1}", 5)
    wait_check("FAKESAT IMAGER COLLECT_TYPE == 'SPECIAL'", 5)
  end
end

class ModeGroup < OpenC3::Group
  def script_safe
    fakesat = FakeSat.new
    fakesat.safe
  end
  def script_checkout
    fakesat = FakeSat.new
    fakesat.checkout
  end
  def script_operate
    fakesat = FakeSat.new
    fakesat.operate
  end
end

class OpsSuite < OpenC3::Suite
  def initialize
    super()
    add_group('CollectGroup')
    add_group('ModeGroup')
  end
end
